src/backend/tools/slack/auth.py

```diff
@@ -0,0 +1,159 @@
+import datetime
+import json
+import urllib.parse
+
+import requests
+from fastapi import Request
+
+from backend.config.settings import Settings
+from backend.crud import tool_auth as tool_auth_crud
+from backend.database_models import ToolAuth
+from backend.database_models.database import DBSessionDep
+from backend.database_models.tool_auth import ToolAuth as ToolAuthModel
+from backend.schemas.tool_auth import UpdateToolAuth
+from backend.services.auth.crypto import encrypt
+from backend.services.logger.utils import LoggerFactory
+from backend.tools.base import BaseToolAuthentication
+from backend.tools.slack.constants import SLACK_TOOL_ID
+from backend.tools.utils.mixins import ToolAuthenticationCacheMixin
+
+logger = LoggerFactory().get_logger()
+
+
+class SlackAuth(BaseToolAuthentication, ToolAuthenticationCacheMixin):
+    TOOL_ID = SLACK_TOOL_ID
+    AUTH_ENDPOINT = "https://slack.com/oauth/v2/authorize"
+    TOKEN_ENDPOINT = "https://slack.com/api/oauth.v2.access"
+    EXCHANGE_ENDPOINT = "https://slack.com/api/oauth.v2.exchange"
+    DEFAULT_BOT_SCOPES = ['search:read.public']
+    DEFAULT_USER_SCOPES = ['search:read']
+
+    def __init__(self):
+        super().__init__()
+
+        settings = Settings()
+        slack_settings = settings.tools.slack if settings.tools and settings.tools.slack else None
+        self.SLACK_CLIENT_ID = getattr(slack_settings, 'client_id', None)
```

Comment:
Prefer: To maintain consistency across the codebase, can you fetch the config with

```
Settings().tools.slack.myconfig
```
For each variable instead? This will also set them as None automatically if it's not populated

Reply:
Pyright checks. We've discussed that